### PR TITLE
fix(bitbucket-server): handle push rejection properly

### LINKED ISSUE #4786: Log better error message when push to Bitbucket Fails
**What would you like Renovate to be able to do?**
I had configured renovate bot incorrectly and when it attempted to push to our internal Bitbucket repository the push would fail, but the error returned by Bitbucket is only logged at a debug level. It made troubleshooting the problem much more difficult. The error you see at INFO level is:

`Repository has changed during renovation - aborting`

But in reality the error I needed to see was:
```
       "err": {
         "message": "remote: You can only push your own commits in this repository        \nremote: Commit 39f0e4633028942919e79b923738149a61ba90a5 was committed by renovate-bot <me@fqdn.com>        \nTo https://**redacted**@fqdn.com/scm/repo.git'\n",
         "stack": "Error: remote: You can only push your own commits in this repository        \nremote: Commit 39f0e4633028942919e79b923738149a61ba90a5 was committed by renovate-bot <me@fqdn.com>        \nTo https://**redacted**@fqdn.com/scm/repo.git'\n\n    at 
```

**Describe the solution you'd like**
I'd really like to see errors coming back from Bitbucket also logged at least at INFO level so I am better able to diagnose the problem. This issue also doesn't result in a failed run of renovate so I didn't notice this issue for several days.

**Describe alternatives you've considered**
None.

**Additional context**
As I said, this issue is for a local Bitbucket server, it's possible this kind of issue could happen with GitHub as well but I don't have personal experience with that.

### PR DESCRIPTION
**Pull request by pret-a-porter**
## Changes:

Git util behaviour patched

## Context:

Fixes: #4786 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/git/index.ts [hunk 1]
```diff
@@ -837,6 +837,13 @@ export async function commitFiles({
       error.validationMessage = `Renovate cannot push to its branch because branch protection has been enabled.`;
       throw error;
     }
+    if (err.message.includes('can only push your own commits')) {
+      const error = new Error(CONFIG_VALIDATION);
+      error.validationSource = branchName;
+      error.validationError = 'Access denied';
+      error.validationMessage = err.message;
```
Comment (rarkins) [inline lib/util/git/index.ts:844]: Let's make it more descriptive

File: lib/util/git/index.ts [hunk 2]
```diff
@@ -837,6 +837,13 @@ export async function commitFiles({
       error.validationMessage = `Renovate cannot push to its branch because branch protection has been enabled.`;
       throw error;
     }
+    if (err.message.includes('can only push your own commits')) {
+      const error = new Error(CONFIG_VALIDATION);
+      error.validationSource = branchName;
+      error.validationError = 'Access denied';
```
Comment (rarkins) [inline lib/util/git/index.ts:843]: 'Bitbucket committer error'

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/util/git/index.ts (hunk 1 above)
Comment (pret-a-porter) [reply lib/util/git/index.ts:844]: 'Renovate cannot push to its branch because has no write access'?

Comment (rarkins) [reply lib/util/git/index.ts:844]: Renovate has experienced the following error when attempting to push its branch to the server: "....."

And then include the error message inside the quotes

**Review by rarkins — APPROVED** [review]

**Pull request merged**